src/test_l3_deepSleepMgr.c

```diff
@@ -0,0 +1,252 @@
+  { "DEEPSLEEPMGR_SET_FAILURE",         (int32_t)DEEPSLEEPMGR_SET_FAILURE         },
+  { "DEEPSLEEPMGR_WAKEUP_FAILURE",      (int32_t)DEEPSLEEPMGR_WAKEUP_FAILURE },
+  { "DEEPSLEEPMGR_TERM_FAILURE",        (int32_t)DEEPSLEEPMGR_TERM_FAILURE        },
+  { "DEEPSLEEPMGR_MAX",                 (int32_t)DEEPSLEEPMGR_MAX                    },
+  {  NULL, -1 }
+};
+
+/**
+ * @brief This function clears the stdin buffer.
+ *
+ * This function clears the stdin buffer.
+ */
+void readAndDiscardRestOfLine(FILE* in)
+{
+   int c;
+   while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/**
+* @brief Initialization of the HAL Deepsleep Manager Module
+*
+* This test provides a scope to open the HAL Deepsleep Manager module.
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 001@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_l3_deepsleep_manager_hal_Init(void)
+{
+   gTestID = 1;
+   DeepSleep_Return_Status_t status = DEEPSLEEPMGR_SUCCESS;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call PLAT_DS_INIT()
+   UT_LOG_INFO("Calling PLAT_DS_INIT()");
+   status = PLAT_DS_INIT();
+   UT_LOG_INFO("Result PLAT_DS_INIT: DeepSleep_Return_Status_t:[%s]",
+                UT_Control_GetMapString(DeepSleep_Return_Status_mapTable, status));
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief Trigger deepsleep from the HAL Deepsleep Manager Module
+*
+* This test provides a scope to trigger deepsleep with a timeout.
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 002@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_deepsleep_manager_hal_Trigger_Deepsleep(void)
+{
+   gTestID = 2;
+   DeepSleep_Return_Status_t status = DEEPSLEEPMGR_SUCCESS;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   int deep_sleep_timeout;
+   bool isGPIOWakeup = false;
+   bool networkStandby = false;
+
+    // Step 1: Get the Deep Sleep Timeout
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("Specify the Deep Sleep Timeout in seconds between 0 and 604800. 0 for no timeout.  ");
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("Select Enable/Disable Wakeup Source: ");
+   scanf("%d", &deep_sleep_timeout);
+   readAndDiscardRestOfLine(stdin);
+   if(deep_sleep_timeout < 0 || deep_sleep_timeout > 604800 )
+        {
+            UT_LOG_ERROR("Invalid timeout value");
+            goto exit;
+        }
+    
+   // Step 2: Call PLAT_DS_SetDeepSleep()
+   UT_LOG_INFO("Calling PLAT_DS_SetDeepSleep()");
+   status = PLAT_DS_SetDeepSleep(deep_sleep_timeout, &isGPIOWakeup, networkStandby);
```

Comment:
please check the return status and assert .

Reply:
Updated assert checks of all the function.